flexbe_core/src/flexbe_core/logger.py

```diff
@@ -34,6 +35,16 @@ def log(text, severity):
         # also log locally
         Logger.local(text, severity)
 
+    @staticmethod
+    def log_throttle(period, text, severity):
+        # creat unique identifier for each logging message
+        log_id = str(severity)  + "_" + text
+        # only log when it's the first time or period time has passed for the logging message
+        if not log_id in Logger._last_logged.keys() or \
+            rospy.Time.now().to_sec() - Logger._last_logged[log_id].to_sec() > period:
+                Logger.log(text, severity)
+                Logger._last_logged.update({log_id: rospy.Time.now()})
```

Comment:
Thanks for adding! Would you see a way that entries can also be removed again eventually so that this dict can't grow infinitely?

Reply:
I think it would be sufficient as a safety-net to just check whether the number of dict entries exceeds a certain limit and then remove entries. As long as the limit is not exceeded, it's probably fine to keep old entries.
As a further idea, I would suggest to log a warning message if the size limit is reached because, as you pointed out, this would likely be caused by improper usage of the throttle such as a non-static messages.